sources/web/datalab/static/datalab.js

```diff
@@ -267,6 +267,29 @@ function initializeNotebookApplication(ipy, notebook, events, dialog, utils) {
     notebook.prototype.save_notebook = function() {
       originalSave.apply(this, /* check_lastmodified */ [ false ]);
     }
+
+    // A replacement notebook copy function that makes copies in the root if
+    // the source is under datalab/.
+    notebook.prototype.copy_notebook = function() {
+      var that = this;
+      var base_url = this.base_url;
+      var w = window.open('', IPython._target);
+      var parent = utils.url_path_split(this.notebook_path)[0];
+      if (parent.startsWith('datalab/')) {
+        parent = '';
```

Comment:
OK, I misread the change. I thought only /datalab was being modified, but the entire path is being removed. So this is fine... but yeah, the inability to specify path when creating copies, and inability to rename across folders leaves something to be desired (and addressed at some point).

For example, maybe this override is where we should prompt the user for a name and optional path.